subsys/bluetooth/host/smp.c

```diff
@@ -1984,6 +1985,14 @@ static int smp_error(struct bt_smp *smp, uint8_t reason)
 {
 	struct bt_smp_pairing_fail *rsp;
 	struct net_buf *buf;
+	bool remote_already_completed;
+
+	/* By spec, SMP "pairing process" completes successfully when the last
+	 * key to distribute is acknowledged at link-layer.
+	 */
+	remote_already_completed = (atomic_test_bit(smp->flags, SMP_FLAG_KEYS_DISTR)
+	                            && !smp->local_dist
+				    && !smp->remote_dist);
```

Comment:
Minor style issue: the line splits should happen right after `&&`, not right before it